android/engine/src/main/java/org/smartregister/fhircore/engine/datastore/ContentCache.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021-2024 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.datastore
```

Comment:
Why are we introducing this cache?

Reply:
There is this Kotlin library https://github.com/MayakaApps/Kache

Reply:
@ellykits I think part of this PR has content from  https://github.com/opensrp/fhircore/pull/3461.  Maybe we can make the switch there cc: @ndegwamartin

Reply:
I think we should rename that class, it is more of a CacheHelper, the actual cache it android native -`androidx.collection.LruCache`

Reply:
updated name